packages/tabs/src/useTabs.ts

```diff
@@ -9,16 +9,46 @@ import { useState } from 'react';
 import { useSelection } from '@zendeskgarden/container-selection';
 import { generateId } from '@zendeskgarden/container-utilities';
 
-function requiredArguments(arg, argStr, methodName) {
+interface IUseTabsPropGetters {
+  getTabProps: <T>(options?: T & IGetTabPropsOptions) => T & IGetTabPropsOptions;
+  getTabPanelProps: <T>(options?: T & IGetTabPanelProps) => IGetTabPanelProps;
+  getTabListProps: <T>(options: T) => T & React.HTMLProps<any>;
+}
+
+interface IGetTabPropsOptions extends React.HTMLProps<any> {
+  index?: number;
+}
+
+interface IGetTabPanelProps extends React.HTMLProps<any> {
+  index?: number;
+  item?: any;
+}
+
+export interface IUseTabsReturnValue extends IUseTabsPropGetters {
+  selectedItem: string;
+  focusedItem: string;
+}
+
+export interface IUseTabsProps extends React.HTMLProps<any> {
+  vertical?: boolean;
+  idPrefix?: string;
+  defaultSelectedIndex?: number;
+}
+
+function requiredArguments(arg: any, argStr: string, methodName: string) {
   if (typeof arg === 'undefined' || arg === null) {
     throw new Error(
       `Accessibility Error: You must provide an "${argStr}" option to "${methodName}()"`
     );
   }
 }
 
-export function useTabs({ vertical, idPrefix, ...options } = {}) {
-  const { selectedItem, focusedItem, getContainerProps, getItemProps } = useSelection({
+export function useTabs({
+  vertical,
+  idPrefix,
+  ...options
+}: IUseTabsProps = {}): IUseTabsReturnValue {
+  const { selectedItem, focusedItem, getContainerProps, getItemProps } = useSelection<any>({
```

Comment:
This should pass `<Item>` to `useSelection` similar to `useButtonGroup`

Reply:
I missed this being new to the concept of generics, but yes! I made `useTabs` a generic hook and it accepts a generic type now that gets passed to `useSelection`. Thank you for the suggestion!